cli/install/install.go

```diff
@@ -0,0 +1,572 @@
+package install
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"regexp"
+	"runtime"
+	"strings"
+
+	"github.com/apex/log"
+	"github.com/tarantool/tt/cli/util"
+)
+
+// FlagInstall is a struct that contains all install flags.
+type FlagInstall struct {
+	InstallReinstall bool
+	InstallForce     bool
+	InstallVerbose   bool
+	InstallNoclean   bool
+}
+
+// Packet is a struct containing sys and install name of the packet.
+type Packet struct {
+	sysName     string
+	installName string
+}
+
+type DistroInfo struct {
+	Name         string
+	Vendor       string
+	Version      string
+	Architecture string
+}
+
+var (
+	PrettyName = regexp.MustCompile(`^PRETTY_NAME=(.*)$`)
+	ID         = regexp.MustCompile(`^ID=(.*)$`)
+	VersionID  = regexp.MustCompile(`^VERSION_ID=(.*)$`)
+)
+
+func GetDistroName() *DistroInfo {
+	var distroInfo DistroInfo
+	if _, err := os.Stat("/lib64/ld-linux-x86-64.so.2"); err == nil {
+		distroInfo.Architecture = "amd64"
+	} else if _, err := os.Stat("/lib/ld-linux.so.2"); err == nil {
+		distroInfo.Architecture = "i386"
+	}
+
+	distoFile, err := os.Open("/etc/os-release")
+	if err != nil {
+		return nil
+	}
+	defer distoFile.Close()
+
+	scanner := bufio.NewScanner(distoFile)
+	for scanner.Scan() {
+		if m := PrettyName.FindStringSubmatch(scanner.Text()); m != nil {
+			distroInfo.Name = strings.Trim(m[1], `"`)
+		} else if m := ID.FindStringSubmatch(scanner.Text()); m != nil {
+			distroInfo.Vendor = strings.Trim(m[1], `"`)
+		} else if m := VersionID.FindStringSubmatch(scanner.Text()); m != nil {
+			distroInfo.Version = strings.Trim(m[1], `"`)
+		}
+	}
+	return &distroInfo
+}
+
+func DetectOsName() string {
+	if runtime.GOOS == "darwin" {
+		return "darwin"
+	}
+	if runtime.GOOS == "windows" {
+		return "windows"
+	}
+	if runtime.GOOS == "linux" {
+		distroInfo := GetDistroName()
+		return distroInfo.Name
+	}
+	return ""
+}
+
+// isDeprecated checks wethere version of programm is below 1.10.0.
+func isDeprecated(version string) bool {
+	splitedVersion := strings.Split(version, ".")
+	if len(splitedVersion) < 2 {
+		return false
+	}
+	if splitedVersion[0] == "1" && len(splitedVersion[1]) < 2 {
+		return true
+	}
+	return false
+}
+
+// getLatestTarantoolTag returns latest tag from tarantool repository.
+func getLatestTarantoolTag() (string, error) {
+	cmd := exec.Command("git", "ls-remote", "--tags", "--sort="+
+		"v:refname", "https://github.com/tarantool/tarantool.git")
```

Comment:
Move `https://github.com/tarantool/tarantool.git` to constant variable, It uses many times.

Reply:
Moved